pkg/providers/azure.go

```diff
@@ -176,26 +179,58 @@ func (sc *SetUpCmd) assignSpRole() error {
 	return nil
 }
 
-func (sc *SetUpCmd) getTenantId() error {
-	log.Debug("Fetching Azure account tenant ID")
-	getTenantIdCmd := exec.Command("az", "account", "show", "--query", "tenantId", "--only-show-errors")
-	out, err := getTenantIdCmd.CombinedOutput()
+func (sc *SetUpCmd) getTenantId(ctx context.Context) error {
+	log.Debug("getting Azure tenant ID")
+
+	tenants, err := listTenants(ctx)
 	if err != nil {
-		log.Printf("%s\n", out)
-		return err
+		return fmt.Errorf("listing tenants: %w", err)
 	}
 
-	var tenantId string
-	if err := json.Unmarshal(out, &tenantId); err != nil {
-		return err
+	if len(tenants) == 0 {
+		return errors.New("no tenants found")
 	}
-	tenantId = fmt.Sprint(tenantId)
-
-	sc.tenantId = tenantId
+	if len(tenants) > 1 {
+		return errors.New("multiple tenants found")
+	}
+	sc.tenantId = fmt.Sprint(&tenants[0])
 
```

Comment:
tenants is a slice of armsubscription.TenantIDDescription, so &tenants[0] is a pointer to a TenantIDDescription. does calling fmt.Sprint(TenantIDDescription) return the TenantID property?

we likely want something similar to `sc.tenantId = *tenants[0].TenantID` to extract the string

Reply:
Addressed the comment in a new PR https://github.com/Azure/draft/pull/259